### ISSUE TITLE
Module 'hashlib' has no '_Hash' member (no-member)

### ISSUE THREAD

**Issue by oittaa** [PyCQA/pylint#5395]
### Bug description

Minimal test case `test.py`:

```
"""docstring"""
from __future__ import annotations
import hashlib

def hash_update(hash_func: hashlib._Hash) -> hashlib._Hash:
    """docstring"""
    hash_func.update(b"0")
    return hash_func

h = hashlib.sha256()
h = hash_update(h)
```

```
$ python3 test.py; echo $?
0
```

```
$ mypy --version; mypy test.py
mypy 0.910
Success: no issues found in 1 source file
```

```
$ pylint --version; pylint test.py
pylint 2.11.1
astroid 2.8.5
Python 3.10.0 (default, Oct 22 2021, 13:24:07) [Clang 13.0.0 (clang-1300.0.29.3)]
************* Module test
test.py:5:27: E1101: Module 'hashlib' has no '_Hash' member (no-member)
test.py:5:45: E1101: Module 'hashlib' has no '_Hash' member (no-member)

--------------------------------------------------------------------
Your code has been rated at -4.29/10 (previous run: -4.29/10, +0.00)
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:5:27: E1101: Module 'hashlib' has no '_Hash' member (no-member)
test.py:5:45: E1101: Module 'hashlib' has no '_Hash' member (no-member)

--------------------------------------------------------------------
Your code has been rated at -4.29/10 (previous run: -4.29/10, +0.00)
```

### Expected behavior

No errors.

### Pylint version

```shell
pylint 2.11.1
astroid 2.8.5
Python 3.10.0 (default, Oct 22 2021, 13:24:07) [Clang 13.0.0 (clang-1300.0.29.3)]
```

### OS / Environment

MacOS

**Comment by Pierre-Sassoulas** [comment]
This is still happening in 2.12.1

**Comment by Pierre-Sassoulas** [comment]
Closing as I can't reproduce on pylint 2.15.0-dev0.
